transform/cast.go

```diff
@@ -82,6 +86,23 @@ func (d *date) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + t.Format(dateOutputFormat) + `"`), nil
 }
 
+// MarshalBSONValue implementa a serialização para o MongoDB
+func (d date) MarshalBSONValue() (bsontype.Type, []byte, error) {
+	t := time.Time(d)
+	return bson.TypeString, bsoncore.AppendString(nil, t.Format(dateOutputFormat)), nil
+}
+
+// UnmarshalBSONValue implementa a desserialização para o MongoDB
```

Comment:
Se for documentar, documente em inglês, mas acho que o o que você adicionou não complementa a semântica.

```suggestion
```